graphics/include/gz/common/Material.hh

```diff
@@ -102,15 +103,22 @@ namespace gz
 
       /// \brief Set a texture image
       /// \param[in] _tex The name of the texture, which must be in the
-      /// resource path
-      public: void SetTextureImage(const std::string &_tex);
+      /// resource path or its name if _img is provided
+      /// \param[in] _img The image containing the texture if image has been
+      /// loaded in memory
+      public: void SetTextureImage(const std::string &_tex,
+                  const std::shared_ptr<Image> &_img = nullptr);
```

Comment:
The main reason is in https://github.com/gazebosim/gz-rendering/pull/642 under "other considerations". When working at the gz-rendering level I noticed that the image gets copied at least twice, once to convert between `gz::common::Image` and `gz::rendering::Image` and once in a gz::rendering::Material::Clone() function, this means that blobs of memory that for each memory that could be several megabytes we will need to do three allocations which might be not great from a performance point of view.
I guess it's a performance vs API clarity issue, I'm really on the fence on this, on one side it is good to avoid premature optimization and having a more obscure API on the other several multi MB copies every time a new model is loaded are not really negligible.